src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs

```diff
@@ -0,0 +1,1277 @@
+                        if (variables.ContainsKey(name))
+                        {
+                            if (currentIndex > variables[name].Item2)
+                            {
+                                variables[name] = new Tuple<string, int, Type>(typeString, currentIndex, variableType);
+                            }
+                        }
+                        else // New type, add it
+                        {
+                            variables.Add(name, new Tuple<string, int, Type>(typeString, currentIndex, variableType));
+                        }
+
+                        continue;
+                    }
+                }
+
+                // Match default types (numbers, dicts, arrays, strings)
+                foreach (var pair in BasicVariableTypes)
+                {
+                    var matches = Regex.Matches(typeString, "^" + pair.Item1 + "$", RegexOptions.Singleline);
+                    if (matches.Count > 0)
+                    {
+                        // If variable has already been encountered
+                        if (variables.ContainsKey(name))
+                        {
+                            if (currentIndex > variables[name].Item2)
+                            {
+                                variables[name] = new Tuple<string, int, Type>(typeString, currentIndex, pair.Item2);
+                            }
+                        }
+                        else // New type, add it
+                        {
+                            variables.Add(name, new Tuple<string, int, Type>(typeString, currentIndex, pair.Item2));
+                        }
+
+                        break;
+                    }
+                }
+            }
+
+            return variables;
+        }
+
+        [Obsolete("Only used to support legacy IronPythonCodeCompletionProvider - remove in 3.0")]
+        /// <summary>
+        /// Attempts to find import statements that look like
+        ///     import lib
+        /// </summary>
+        /// <param name="code">The code to search</param>
+        /// <returns>A dictionary matching the lib to the code where lib is the library being imported from</returns>
+        internal static Dictionary<string, string> FindBasicImportStatements(string code)
+        {
+            var pattern = "^" + basicImportRegex + spacesOrNone + variableName;
+
+            var matches = Regex.Matches(code, pattern, RegexOptions.Multiline);
+
+            var importMatches = new Dictionary<string, string>();
+
+            for (var i = 0; i < matches.Count; i++)
+            {
+                var wholeLine = matches[i].Groups[0].Value;
+                var libName = matches[i].Groups[3].Value.Trim();
+
+                if (importMatches.ContainsKey(libName))
+                {
+                    continue;
+                }
+
+                importMatches.Add(libName, wholeLine);
+            }
+
+            return importMatches;
+        }
+
```

Comment:
I think the `#endregion` for the `#region backup legacy class` should come somewhere here and not line 904, correct?